image_proc/CMakeLists.txt

```diff
@@ -1,68 +1,51 @@
-
-catkin_package(
-  CATKIN_DEPENDS image_geometry roscpp sensor_msgs
-  DEPENDS OpenCV
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-)
-
-include_directories(include ${catkin_INCLUDE_DIRS} ${OpenCV_INCLUDE_DIRS})
-
-# Temporary fix for DataType deprecation in OpenCV 3.3.1. We continue to use the deprecated form for now because
-# the new one is not available in OpenCV 2.4 (on Trusty).
-add_definitions(-DOPENCV_TRAITS_ENABLE_DEPRECATED)
-
-# Nodelet library
-add_library(${PROJECT_NAME} src/libimage_proc/processor.cpp
-                                src/nodelets/debayer.cpp
-                                src/nodelets/rectify.cpp
-                                src/nodelets/resize.cpp
-                                src/nodelets/crop_decimate.cpp
-                                src/libimage_proc/advertisement_checker.cpp
-                                src/nodelets/edge_aware.cpp
-                                src/nodelets/crop_non_zero.cpp
-)
-add_dependencies(${PROJECT_NAME} ${PROJECT_NAME}_gencfg)
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${Boost_LIBRARIES} ${OpenCV_LIBRARIES})
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-install(TARGETS ${PROJECT_NAME}
-        DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-)
-install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-)
-install(FILES nodelet_plugins.xml
-        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-)
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
+find_package(cv_bridge REQUIRED)
+find_package(image_transport REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(OpenCV REQUIRED)
+find_package(image_geometry REQUIRED)
+
+include_directories(include)
+add_executable(image_proc
+               src/image_proc.cpp
+               src/debayer.cpp
+               src/rectify.cpp
+               src/edge_aware.cpp)
+target_link_libraries(image_proc ament_index_cpp::ament_index_cpp)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_link_libraries(image_proc "stdc++fs")
+endif()
 
-# Standalone node
-add_executable(image_proc_exe src/nodes/image_proc.cpp)
-target_link_libraries(image_proc_exe ${PROJECT_NAME}  ${Boost_LIBRARIES} ${OpenCV_LIBRARIES})
-SET_TARGET_PROPERTIES(image_proc_exe PROPERTIES OUTPUT_NAME image_proc)
-install(TARGETS image_proc_exe
-        DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+ament_target_dependencies(image_proc
+  "image_geometry"
+  "sensor_msgs"
+  "cv_bridge"
+  "class_loader"
+  "image_transport"
+  "rclcpp"
+  "rclcpp_components"
+  "rcutils"
 )
 
-# install the launch file
-install(DIRECTORY launch
-        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/
-)
 
-if(CATKIN_ENABLE_TESTING)
-  add_subdirectory(test)
-endif()
+install(TARGETS
+        image_proc
+        DESTINATION bin/${PROJECT_NAME})
```

Comment:
I think this might be lib/ , but I'm open for suggestions. I can't seem to get the executable to be visible if it's in the bin folder.

Reply:
Go ahead with `lib/`. It seems that my `bin/` recommendation was either wrong or out-of-date.